src/include/sof/coherent.h

```diff
@@ -80,9 +80,23 @@ struct coherent {
 
 /* debug sharing amongst cores */
 #ifdef COHERENT_CHECK_NONSHARED_CORES
+
+#define CORE_CHECK_STRUCT_FIELD uint32_t __core; bool __is_shared
+#define CORE_CHECK_STRUCT_INIT(_c, is_shared) { (_c)->__core = cpu_get_id(); \
+						(_c)->__is_shared = is_shared; }
+#define CORE_CHECK_STRUCT(_c) { assert(!!(_c)->__is_shared == !!is_uncached(_c)); \
+				assert(cpu_get_id() == (_c)->__core || (_c)->__is_shared); }
```

Comment:
I'd put all the conditions in a single `assert()`

Reply:
@lyakh Given this is a debug option, woudln't separate asserts be nicer? You'd see which condition failed right away.

Reply:
@kv2019i , should add COHERENT_CHECK_NONSHARED_CORES to debug build?

Reply:
agree with @lyakh, 2 asserts are also more clear to read
As it is a debug option it does make any changes in production performance or code size

Reply:
> @lyakh Given this is a debug option, woudln't separate asserts be nicer? You'd see which condition failed right away.

@kv2019i if it was spelled out in the code directly, then yes, separate asserts would be better. Here it's a single macro, so I thought using a single assert would fit better. But yes, you're right, with separate ones you get precise diagnostics immediately.

Reply:
@marcinszkudlinski , please check code style, seems _c is not a good name.